### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by hudochenkov**
Updated all dependencies prior release.

Didn't update TypeScript, because it has a bug https://github.com/microsoft/TypeScript/issues/34869.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -138,7 +138,7 @@
     "lint:types": "tsc",
     "lint": "npm-run-all --parallel lint:*",
     "pretest": "npm-run-all --serial lint flow prettier:check",
-    "prettier:check": "prettier \"**/*.js\" --list-different",
+    "prettier:check": "prettier \"**/*.js\" --check",
```
Comment (hudochenkov) [inline package.json]: Will give human-friendly status messages.

https://prettier.io/docs/en/cli.html#check

**Comment by ybiquitous** [comment]
I see the following packages should not be updated:
- `typescript` ... see https://github.com/stylelint/stylelint/pull/4404#issue-339016501
- `globby` ... see #4254

However, should we update other packages?

```
$ npm outdated
Package                  Current   Wanted   Latest  Location
flow-bin                 0.100.0  0.100.0  0.111.3  stylelint
globby                     9.2.0    9.2.0   10.0.1  stylelint
jest-watch-typeahead       0.4.0    0.4.2    0.4.2  stylelint
postcss-sass               0.4.1    0.4.2    0.4.2  stylelint
postcss-selector-parser    3.1.1    3.1.1    6.0.2  stylelint
remark-cli                 7.0.0    7.0.1    7.0.1  stylelint
string-width               4.1.0    4.2.0    4.2.0  stylelint
typescript                 3.6.4    3.7.2    3.7.2  stylelint
```

**Comment by hudochenkov** [comment]
There are few new versions since my last commit.

`flow-bin` we don't update, because we're moving to TypeScript.

`postcss-selector-parser` has own PR.

Other I'll update.

**Comment by ybiquitous** [comment]
> `postcss-selector-parser` has own PR.

I missed #3988. Thanks. 👍

**Pull request merged**